src/storage/store/var_list_column_chunk.cpp

```diff
@@ -27,35 +27,57 @@ void VarListDataColumnChunk::resizeBuffer(uint64_t numValues) {
 
 VarListColumnChunk::VarListColumnChunk(
     LogicalType dataType, uint64_t capacity, bool enableCompression, bool inMemory)
-    : ColumnChunk{std::move(dataType), capacity, enableCompression, true /* hasNullChunk */},
-      needFinalize{false} {
+    : ColumnChunk{std::move(dataType), capacity, enableCompression, true /* hasNullChunk */} {
+    sizeColumnChunk = ColumnChunkFactory::createColumnChunk(
+        *common::LogicalType::UINT32(), enableCompression, capacity);
     varListDataColumnChunk = std::make_unique<VarListDataColumnChunk>(
         ColumnChunkFactory::createColumnChunk(*VarListType::getChildType(&this->dataType)->copy(),
             enableCompression, 0 /* capacity */, inMemory));
+    checkOrder = true;
     KU_ASSERT(this->dataType.getPhysicalType() == PhysicalTypeID::VAR_LIST);
 }
 
+bool VarListColumnChunk::isOffsetSortedAscending(uint64_t startPos, uint64_t endPos) const {
```

Comment:
Re-read on this, actually this function is not just check sorting order, but also checking if offsets are consecutive or not.
So the name is confusing. we should rename it. `isOffsetsConsecutiveAndSortedAscending` is a candidate.